src/contracts/account/RelayHub.sol

```diff
@@ -3,78 +3,83 @@ pragma experimental ABIEncoderV2;
 
 import "@openzeppelin/contracts/utils/Create2.sol";
 import "./ReplayProtection.sol";
+import "./ContractCall.sol";
 
 /**
  * A minimal relay hub contract.
  * Verifies the signer's signature and replay protection before forwarding calldata to the target.
  * Delegates nonce verification to another contract.
+ * Note it does NOT support delegatecall to avoid memory corruption problems.
  */
-contract RelayHub is ReplayProtection {
+contract RelayHub is ReplayProtection, ContractCall {
 
     event Deployed(address signer, address addr);
-    event Revert(string reason);
+
+    struct MetaTx {
+        address target;
```

Comment:
mentioned this on the chat earlier. so when we sign and encode a meta-tx, the proxycontract/relay hub is set to "to" and the "target" contract is what we are calling.

Since we are no longer surfacing ForwardParams. We could potentially change it. But there might still be some confusion (e.g. a minimalTx.to is the proxy contract/relay hub).